shell/common/api/atom_api_native_theme.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/common/api/atom_api_native_theme.h"
+
+#include "native_mate/dictionary.h"
+#include "native_mate/object_template_builder.h"
+#include "shell/common/node_includes.h"
+#include "ui/gfx/color_utils.h"
+#include "ui/native_theme/native_theme.h"
+
+namespace electron {
+
+namespace api {
+
+NativeTheme::NativeTheme(v8::Isolate* isolate, ui::NativeTheme* theme)
+    : theme_(theme) {
+  theme_->AddObserver(this);
+  Init(isolate);
+}
+
+NativeTheme::~NativeTheme() {
+  theme_->RemoveObserver(this);
+}
+
+void NativeTheme::OnNativeThemeUpdated(ui::NativeTheme* theme) {
+  Emit("updated");
+}
+
+bool NativeTheme::ShouldUseDarkColors() {
+  return theme_->SystemDarkModeEnabled();
+}
+
+bool NativeTheme::ShouldUseHighContrastColors() {
+  return theme_->UsesHighContrastColors();
+}
+
+// static
+v8::Local<v8::Value> NativeTheme::Create(v8::Isolate* isolate) {
+  ui::NativeTheme* theme = ui::NativeTheme::GetInstanceForNativeUi();
+  return mate::CreateHandle(isolate, new NativeTheme(isolate, theme)).ToV8();
+}
+
+// static
+void NativeTheme::BuildPrototype(v8::Isolate* isolate,
+                                 v8::Local<v8::FunctionTemplate> prototype) {
+  prototype->SetClassName(mate::StringToV8(isolate, "NativeTheme"));
+  mate::ObjectTemplateBuilder(isolate, prototype->PrototypeTemplate())
+      .SetProperty("shouldUseDarkColors", &NativeTheme::ShouldUseDarkColors)
+      .SetProperty("shouldUseHighContrastColors",
+                   &NativeTheme::ShouldUseHighContrastColors)
+      .SetProperty("shouldUseInvertedColorScheme",
```

Comment:
In favor of consistency, imo there should be `NativeWindow::ShouldUseInvertedColorScheme` as well

Reply:
There is no need for this (it would just be an empty method that called another method, more efficient to bind directly to the target)